code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/PropositionInfo.java

```diff
@@ -51,14 +54,14 @@ private PropositionInfo(final Map<String, Object> propositionInfoMap) {
         }
     }
 
-    static PropositionInfo create(final Map<String, Object> propositionInfoMap) {
-        if (StringUtils.isNullOrEmpty(DataReader.optString(propositionInfoMap, ID, ""))) {
+    static PropositionInfo create(final Map<String, Object> propositionInfoMap) throws Exception {
+        if (StringUtils.isNullOrEmpty(DataReader.getString(propositionInfoMap, ID))) {
             return null;
```

Comment:
can you reach these return calls?  if we are passing along the exception, then we'll never get to them, right?